Log explorer attribute search (#3838)
DataDog/documentation: content/logs/explorer/search.md

| `OR`         | **Union**: either term is contained in the selected events                                             | authentication OR password   |
| `-`          | **Exclusion**: the following term is NOT in the event                                                  | authentication AND -password |

### Autocomplete

Use the search bar's autocomplete feature to complete your query using existing values:

{{< img src="logs/explorer/search/search_bar_autocomplete.png" alt="search bar autocomplete " responsive="true" style="width:80%;">}}

### Escaping of special characters

The following characters are considered special: `?`, `>`, `<`, `:`, `=`,`"`, `~`, `/`, and `\` require escaping with the `\` character.

### Attributes search

> ### Message attribute search

Reviewer: ```suggestion
#### Message attribute search
```

> To search logs that contain `user=12345` in their message attribute the following search must be entered:

Reviewer: ```suggestion
To search for logs that contain `user=12345` in the message attribute use the following search:
```

`@url:www.datadoghq.com`

**Note**: Searching on a facet value that contains special characters requires escaping or double quotes. The same logic is be applied to spaces within log attributes. Log attributes should not contain spaces, but if they do, spaces must be escaped. If an attribute is called `user.first name`, perform a search on this attribute by escaping the space: `@user.first\ name:myvalue`

Examples:

| Search query                                                         | Description                                                                                                                                                         |
| `@http.url_details.path:"/api/v1/test"`                              | Searches all logs containing `/api/v1/test` in the attribute `http.url_details.path`.                                                                               |
| `@http.url:\/api\/v1\/*`                                             | Searches all logs containing a value in `http.url` attribute that start with `/api/v1/`                                                                             |

> | `@http.status_code:[200 TO 299] @http.url_details.path:\/api\/v1\/*` | Searches all logs containing a `http.status_code` value between 200 and 299, and containing a value in `http.url_details.path` attribute that start with `/api/v1/` |

Reviewer: The above doesn't display as a table in the preview
